blocks/library/quote/index.js

```diff
@@ -120,7 +120,7 @@ registerBlock( 'core/quote', {
 					onMerge={ mergeWithPrevious }
 					showAlignments
 				/>
-				{ ( citation || !! focus ) && (
+				{ ( ( citation && citation.length > 0 ) || !! focus ) && (
```

Comment:
Preempting a concern here: Yes, this is ugly. Rather than extract to a separate variable, I think we should acknowledge its ugliness and try to avoid it altogether by providing default values for children. This should be addressed separately.